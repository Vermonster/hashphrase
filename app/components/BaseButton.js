import React from 'react';
import { Button, Text } from 'react-native-paper';
import PropTypes from 'prop-types';

const BaseButton = ({ onPress, buttonText }) => (
  <Button mode="contained" onPress={onPress}>
    <Text>{ buttonText }</Text>
  </Button>
);

BaseButton.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default BaseButton;
