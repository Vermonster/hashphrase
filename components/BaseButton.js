import React from 'react';
import { Button, Text } from 'native-base';
import PropTypes from 'prop-types';

const BaseButton = ({onPress, buttonText}) => (
  <Button full onPress={onPress}>
    <Text>{buttonText}</Text>
  </Button>
);

BaseButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default BaseButton;