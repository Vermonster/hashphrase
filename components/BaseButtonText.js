import React from 'react';
import { Text } from 'native-base';

const BaseButtonText = (props) => (
    <Text>{props.label}</Text>
);

export default BaseButtonText;