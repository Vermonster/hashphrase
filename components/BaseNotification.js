import React from 'react';
import {
  Card, CardItem, Body, Text,
} from 'native-base';

const BaseNotification = ({ message }) => (
  <Card>
    <CardItem>
      <Body>
        <Text>
          { message }
        </Text>
      </Body>
    </CardItem>
  </Card>
);

export default BaseNotification;
