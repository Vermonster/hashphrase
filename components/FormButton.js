import React from 'react';
import { Button, Text } from 'native-base';
import PropTypes from 'prop-types';

const FormButton = (props) => (
  <Button full onPress={props.onPress}>
    <Text>Generate Password</Text>
  </Button>
);

FormButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default FormButton;