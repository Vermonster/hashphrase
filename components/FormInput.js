import React from 'react';
import { Input } from 'native-base';
import PropTypes from 'prop-types';

const FormInput = ({
  name, placeholder, secureTextEntry, onChange,
}) => (
  <Input
    name={name}
    placeholder={placeholder}
    secureTextEntry={secureTextEntry}
    autoCapitalize="none"
    onChange={e => onChange(e)}
  />
);

FormInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
};

export default FormInput;
