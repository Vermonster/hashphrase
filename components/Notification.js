import React from 'react';
import { Card, CardItem, Body, Text } from 'native-base';

const Notification = (props) => (
  <Card>
    <CardItem>
      <Body>
        <Text>
          {props.text}
        </Text>
      </Body>
    </CardItem>
  </Card>
);

export default Notification;