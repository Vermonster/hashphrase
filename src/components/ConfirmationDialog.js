import React from 'react';
import { withNamespaces } from 'react-i18next';
import {
  View, Text, Modal, StyleSheet, Clipboard, TextInput, Platform, Dimensions,
} from 'react-native';
import { Button, IconButton } from 'react-native-paper';
import { colors, fontSize } from '../styles/base';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalBackground: {
    backgroundColor: colors.modalBackground,
  },
  innerContainer: {
    backgroundColor: colors.white,
    // flex: 1 / 2,
    borderRadius: 4,
  },
  messagesContainer: {
    flex: 1 / 3,
    backgroundColor: colors.primary,
    borderTopRightRadius: 4,
    borderTopLeftRadius: 4,
    justifyContent: 'center',
  },
  passwordContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    alignSelf: 'center',
    width: '95%',
    backgroundColor: colors.secondary,
    borderRadius: 4,
    padding: 6,
  },
  password: {
    flex: 1,
    justifyContent: 'space-around',
    marginLeft: '5%',
  },
  title: {
    color: colors.secondary,
    fontSize: fontSize.xxl,
    marginBottom: '3%',
  },
  paragraph: {
    fontSize: fontSize.lg,
  },
  label: {
    fontSize: fontSize.lg,
    color: colors.white,
  },
  buttonRow: {
    justifyContent: 'flex-start',
    alignSelf: 'center',
    flex: 1 / 3,
    // width: 'auto', SET WIDTH % BASED ON OVERALL WIDTH
  },
  button: {
    borderColor: colors.secondary,
    borderWidth: 2,
  },
  closeButton: {
    alignSelf: 'flex-end',
    top: '1%',
    right: '0.5%',
    position: 'absolute',
  },
});

class ConfirmationDialog extends React.Component {
  state = {
    obscured: true,
    window: Dimensions.get('window'),
  }

  componentWillMount = () => {
    Dimensions.addEventListener('change', this.handler);
  }

  componentWillUnmount() {
    Dimensions.removeEventListener('change', this.handler);
  }

  handler = dims => this.setState(dims);

  handleClearClipboard = () => Clipboard.setString('');

  toggleObscured = () => {
    this.setState(prevState => ({ obscured: !prevState.obscured }));
  };

  handleSubmit = () => {
    const { closeModal, showSnackbar, clearForm } = this.props;
    this.handleClearClipboard();
    showSnackbar();
    this.resetState();
    clearForm();
    return closeModal();
  }

  resetState = () => {
    this.setState({ obscured: true });
  }

  modalStyling = (orientation, height, width) => {
    const styling = {};
    // if (orientation === 'landscape') {
    if (width < 600) {
      styling.innerContainerWidth = '95%';
    } else {
      styling.innerContainerWidth = '70%';
    }
    if (height > 800) {
      styling.innerContainerFlex = 1 / 2;
    } else if (height < 350) {
      styling.innerContainerFlex = 4 / 5;
    } else {
      styling.innerContainerFlex = 2 / 3;
    }
    // } else if (orientation === 'portrait') {
    //   styling.inputHt = 1 / 3;
    // }
    return styling;
  }

  render() {
    const { obscured, window } = this.state;
    const { width, height } = window;
    const orientation = height > width ? 'portrait' : 'landscape';
    const {
      t, visible, generatedPassword, closeModal, clearForm,
    } = this.props;
    const visibilityIcon = obscured ? 'visibility-off' : 'visibility';
    const accountLabelMargin = Platform.OS === 'ios' ? (0.02 * height) : 0;
    const dynamicStyles = this.modalStyling(orientation, height, width);
    const { innerContainerWidth, innerContainerFlex } = dynamicStyles;
    console.log(height, width);

    return (
      <View
        style={styles.container}
      >
        <Modal
          animationType="fade"
          transparent
          visible={visible}
          onRequestClose={closeModal}
          supportedOrientations={['portrait', 'portrait-upside-down', 'landscape', 'landscape-left', 'landscape-right']}
        >
          <View style={[styles.container, styles.modalBackground]}>
            <View style={[styles.innerContainer, { width: innerContainerWidth, flex: innerContainerFlex }]}>
              <View style={styles.messagesContainer}>
                <IconButton
                  icon="close"
                  size={35}
                  style={styles.closeButton}
                  onPress={() => { closeModal(); clearForm(); }}
                />
                <View style={{ marginLeft: '4%' }}>
                  <Text style={styles.title}>{t('completedStatus')}</Text>
                  <Text style={[styles.paragraph, { color: colors.white, width: '80%' }]}>
                    {t('completedClipboard')}
                  </Text>
                </View>
              </View>
              <View style={{ flex: 1 / 2, justifyContent: 'center' }}>
                <View style={[styles.passwordContainer]}>
                  <View style={styles.password}>
                    <Text style={[styles.label, { marginBottom: accountLabelMargin }]}>
                      {t('accountPassword')}
                    </Text>
                    <TextInput
                      secureTextEntry={obscured}
                      editable={false}
                      style={styles.label}
                    >
                      { generatedPassword }
                    </TextInput>
                  </View>
                  <IconButton
                    icon={visibilityIcon}
                    color={colors.white}
                    onPress={this.toggleObscured}
                    style={styles.hideShowButton}
                  />
                </View>
              </View>
              <View style={styles.buttonRow}>
                <Button
                  testID="submit-form"
                  onPress={this.handleSubmit}
                  accessibilityLabel="ACTION BUTTON"
                  style={styles.button}
                  mode="outlined"
                  color={colors.secondary}
                >
                  <Text>{t('button')}</Text>
                </Button>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    );
  }
}

export default withNamespaces('confirmationDialog')(ConfirmationDialog);
export { ConfirmationDialog as TestConfirmationDialog };
