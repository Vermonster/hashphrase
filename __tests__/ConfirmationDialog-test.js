import React from 'react';
import { shallow } from 'enzyme';
import { Checkbox, Button } from 'react-native-paper';
import { TestConfirmationDialog as ConfirmationDialog } from '../src/components/ConfirmationDialog';

describe('<ConfirmationDialog />', () => {
  let wrapper;
  beforeEach(() => {
    const mockFn = jest.fn();
    wrapper = shallow(
      <ConfirmationDialog
        clearClipboard={false}
        closeModal={mockFn}
      />,
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state when clear checkbox is clicked', () => {
    wrapper.find(Checkbox).simulate('press');
    expect(wrapper.state('clearClipboard')).toBe(true);
  });

  it('should clear clipboard when the clear clipboard checkbox is clicked', () => {
    wrapper.setState({ clearClipboard: true });
    const spy = jest.spyOn(wrapper.instance(), 'handleClearClipboard');
    wrapper.find(Button).simulate('press');
    expect(spy).toBeCalled();
  });
});
