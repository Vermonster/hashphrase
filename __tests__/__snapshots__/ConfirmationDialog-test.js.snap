// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationDialog /> should match the snapshot 1`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[MockFunction]}
    supportedOrientations={
      Array [
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <ForwardRef(View)
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
          },
        ]
      }
    >
      <ForwardRef(View)
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 4,
              "paddingHorizontal": "5%",
            },
            Object {
              "flex": 0.2,
              "width": "60%",
            },
          ]
        }
      >
        <ForwardRef(View)
          style={
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "flex": 0.5,
              "paddingTop": "5%",
            }
          }
        >
          <ForwardRef(View)>
            <withTheme(Text)
              style={
                Object {
                  "color": "#E07926",
                  "fontFamily": "lato-bold",
                  "fontSize": 24,
                  "marginBottom": "3%",
                }
              }
            >
              completedStatus
            </withTheme(Text)>
            <withTheme(Text)
              style={
                Array [
                  Object {
                    "fontSize": 16,
                  },
                ]
              }
            >
              completedClipboard
            </withTheme(Text)>
          </ForwardRef(View)>
        </ForwardRef(View)>
        <ForwardRef(View)
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#212A59",
                "borderRadius": 4,
                "borderWidth": 1,
                "flex": 0.3333333333333333,
                "flexDirection": "row",
              },
              Object {
                "width": undefined,
              },
            ]
          }
        >
          <TextInput
            accessibilityLabel="Generated password"
            allowFontScaling={true}
            editable={false}
            secureTextEntry={true}
            style={
              Object {
                "color": "#212A59",
                "flex": 1,
                "fontFamily": "lato-regular",
                "fontSize": 16,
                "paddingLeft": "3%",
              }
            }
            underlineColorAndroid="transparent"
          >
            12345678
          </TextInput>
          <withTheme(IconButton)
            color="#212A59"
            icon="visibility-off"
            onPress={[Function]}
          />
        </ForwardRef(View)>
        <ForwardRef(View)
          style={
            Object {
              "alignSelf": "flex-end",
              "flex": 0.5,
              "justifyContent": "center",
            }
          }
        >
          <withTheme(Button)
            accessibilityLabel="Press to start over"
            contentStyle={
              Object {
                "width": "auto",
              }
            }
            mode="text"
            onPress={[Function]}
            testID="start-over-button"
            theme={
              Object {
                "colors": Object {
                  "text": "#212A59",
                },
                "fonts": Object {
                  "regular": "lato-bold",
                },
              }
            }
          >
            <withTheme(Text)>
              button
            </withTheme(Text)>
          </withTheme(Button)>
        </ForwardRef(View)>
      </ForwardRef(View)>
    </ForwardRef(View)>
  </Component>
</ForwardRef(View)>
`;
